typescript/helpers.d.ts

```diff
@@ -12,4 +12,4 @@ export type $OmitArrayKeys<Arr> = Arr extends readonly any[] ? Omit<Arr, keyof a
 
 export type $PreservedValue<Value, Fallback> = [Value] extends [never] ? Fallback : Value;
 
-export type $NormalizeIntoArray<T extends unknown | unknown[]> = T extends unknown[] ? T : [T];
+export type $NormalizeIntoArray<T extends unknown | readonly unknown[]> = T extends readonly unknown[] ? T : [T];
```

Comment:
With this fix when using a normal array as `defaultNS` the types won't work. 
You can try out removing `Readonly<...> `from the `CustomTypeOptions.defaultNS.

You have to take into account both array types. 

This should do the trick:
```suggestion
export type $NormalizeIntoArray<T extends unknown | readonly unknown[] | unknown[]> = T extends
  | unknown[]
  | readonly unknown[]
  ? T
  : [T];
```

Reply:
What version of TS did you test this with? I tested both variants and they seem to work. See this playground https://www.typescriptlang.org/play?#code/C4TwDgpgBAkgzgQQE5IIYgDwBUB8UC8UWUEAHsBAHYAmcUSEq1A9pQDYhQCulA1pcwDulANoBdKAH4owJF2gAuKADNUbOBABQmgMas4weo2oB5dp0IiAjABoATDYDME1HT2UDu-YYFIAtmoEUNb2TmLaoJBQqEHwyGiYkRDMykZMZhw4mgD02VBQAHqSmklQAEaxiCjoGEkpUL4BbFm5+UVAA

Reply:
You are right, sorry.
I tested this on my local branch but probably I had something on my working copy that broke the test 😅